<?xml version="1.0"?>
<launch>

    <!-- Launch sick_lidar3d -->
    <!-- env name="ROSCONSOLE_CONFIG_FILE" value="/tmp/rosconsole_loglevel_warn.conf" / -->
    <arg name="mrs100_ip" default="192.168.0.1"/> <!-- IP address of multiScan136, overwrites default ip address "192.168.0.1" in sick_lidar3d.yaml -->
    <arg name="mrs100_dst_ip" default=""/>        <!-- UDP destination IP address (ip address of udp receiver), overwrites default in sick_lidar3d.yaml -->
    <arg name="mrs100_post_start_stop" default="False"/> <!-- Post start and stop commands to start/stop multiScan136 using Rest-API -->
    
    <rosparam command="load" file="$(find sick_scan)/launch/sick_lidar3d.yaml" />
    <param name="/sick_lidar3d/ros__parameters/mrs100_ip" value="$(arg mrs100_ip)" /> <!-- IP address of multiScan136 to post start and stop commands using Rest-API, f.e. "192.168.0.1" (default) -->
    <param name="/sick_lidar3d/ros__parameters/mrs100_dst_ip" value="$(arg mrs100_dst_ip)" /> <!-- UDP destination IP address (ip address of udp receiver), f.e. "192.168.0.140" -->
    <param name="/sick_lidar3d/ros__parameters/mrs100_post_start_stop" value="$(arg mrs100_post_start_stop)" /> <!-- Post start and stop commands to start/stop multiScan136 using Rest-API -->

    <node name="lidar3d_mrs100_recv" pkg="sick_scan" type="lidar3d_mrs100_recv" output="screen">
        <param name="scanner_type" type="string" value="sick_lidar3d_mrs100"/>
    </node>

</launch>

